edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/identity/DeviceIdentity.cs

```diff
@@ -1,27 +1,59 @@
 // Copyright (c) Microsoft. All rights reserved.
 namespace Microsoft.Azure.Devices.Edge.Hub.Core.Identity
 {
-    using System;
     using Microsoft.Azure.Devices.Edge.Hub.Core.Device;
     using Microsoft.Azure.Devices.Edge.Util;
 
     public class DeviceIdentity : Identity, IDeviceIdentity
     {
-        readonly Lazy<string> asString;
-
         public DeviceIdentity(
             string iotHubHostName,
+            Option<string> gatewayHostname,
             string deviceId)
-            : base(iotHubHostName)
+            : base(iotHubHostName, gatewayHostname, deviceId)
         {
             this.DeviceId = Preconditions.CheckNonWhiteSpace(deviceId, nameof(deviceId));
-            this.asString = new Lazy<string>(() => $"DeviceId: {this.DeviceId} [IotHubHostName: {this.IotHubHostName}]");
         }
 
         public string DeviceId { get; }
 
-        public override string Id => this.DeviceId;
+        protected bool Equals(DeviceIdentity other) =>
```

Comment:
@varunpuranik would you tell me why we don't override Equality members in DeviceIdentity and ModuleIdentity before?  thanks.

Reply:
coz we havne't needed to. We just compare Ids.